### ISSUE TITLE
ACDC结果

### ISSUE THREAD

**Issue by fakerlove1** [ziyangwang007/CV-SSL-MIS#3]
ACDC 20% label dice能到91%吗？我训练只能到88%多

**Comment by ziyangwang007** [comment]
Thanks for your interest and testing.

I personally reckon that the performance may be higher or lower each time, because of (1) the hyperparameter setting (2)loss function design, especially the weight setting between dice loss, ce loss, or supervision loss and semi-supervision loss (3) the random selection of labeled train, unlabeled train, validation, and testing, (4) the size of images, 256, 224, or the original size of your training/testing set

If your evaluation results are a little bit lower than the reported number on the paper, I believe that the performance of other baseline methods should also little bit lower than the reported number on the paper, if you use the same setting including loss function, feature distribution, random seed, and etc.

Some baseline methods with the same setting will be released soon this year.

**Comment by smallkaka** [comment]
> ACDC 20% label dice能到91%吗？我训练只能到88%多

Hello, did you solve it? I trained only 87.6

**Comment by fakerlove1** [comment]
> > > ACDC 20% label dice能到91%吗？我训练只能到88%多
> > 
> > 
> > Hello, did you solve it? I trained only 87.6
> 
> When I use the settings in the paper, such as bacth_size=24, lr=0.0001, I get a lower score of dice=66.7

额，这样子的。模型我换成 segformer-b1。没有使用 swinunet。表现太差了。
参数配置是这样子的。
vit模型基本上都得用adam,或者adamw配置。
结果是unet 89.19。segformer的dice大概在87左右。

`# 数据集配置
datasets: "acdc"
num_classes: 4
data_path: "/home/ubuntu/data/ACDC"
save_path: "checkpoint/2023-01-27-S4CVnet-ACDC"
name: "CTCT-ACDC"
ckpt: None
cuda: True

train_crop_size: [224,224]
test_crop_size: [224,224]
crop_val: False
batch_size: 8
unlabel_batch_size: 24
num_workers: 4

seed: 1337
label_num: 0.1
consistency: 0.1
consistency_rampup: 200.0
ema_decay: 0.99
total_itrs: 30000
step_size: 1500

模型配置
model1:
model: "unet"
variant: "unet"
in_channels: 1
num_classes: 4

优化器训练配置
opt: "sgd"
lr: 0.01
weight_decay: 0.0005
momentum: 0.9
sched: "medical"
warmup_epochs: 1
warmup_lr: 0.0001
cooldown_epochs: 10
min_lr: 0.000001
total_itrs: 30000
step_size: 1500

model2:
model: "segformer"
variant: "segformer"
in_channels: 1
num_classes: 4
train_crop_size: [224,224]

优化器训练配置
opt: "adamW"
lr: 0.0008
weight_decay: 0.05
momentum: 0.9
sched: "medical"
warmup_epochs: 1
warmup_lr: 0.00001
cooldown_epochs: 10
min_lr: 0.000001
total_itrs: 30000
step_size: 1500

`

**Comment by fakerlove1** [comment]
https://github.com/ziyangwang007/CV-SSL-MIS/issues/3#issuecomment-1504887924

大概率是这个。

`def calculate_metric_percase(pred, gt):
    pred[pred > 0] = 1
    gt[gt > 0] = 1
    if pred.sum() > 0:
        dice = metric.binary.dc(pred, gt)
        hd95 = metric.binary.hd95(pred, gt)
        return dice, hd95

    elif pred.sum() > 0 and gt.sum() == 0:
        return 1, 0
    else:
        return 0, 0`

**Comment by ziyangwang007** [comment]
Hi,

Thank you, @smallkaka and @fakerlove1, for your interest in trying to run the code.

Thank you for pointing out that S4CVNet did not achieve a dice score of 0.91 on your machine. I have retested all baseline networks and S4CVNet with your settings, and S4CVNet still achieves state-of-the-art performance against all other baseline methods.

Different people may report different numbers with different settings. Even with only 10% of the labeled training set, there can still be differences if you randomly select the data each time. Additionally, the weight loss function may have different settings, such as the weight between dice and cross entropy or between different supervision and semi-supervision losses.

I suggest that you run all experiments with your hyperparameter settings. If one network performs slightly worse than reported in the paper, I believe that all other baseline methods (such as mean teacher and uncertainty-aware mean teacher) will also perform slightly worse than their reported numbers. Therefore, you should not focus on the difference between your results and the reported numbers, but rather on the difference between your results and the numbers of other baseline methods with the same settings.

Please stay in touch. I have recently reproduced some other baseline methods, such as Contrastive Learning, MoCo,and etc and will be adding about five more new baseline methods to this GitHub repository in next two months.

Best regards.

**Comment by ziyangwang007** [comment]
I have reproduced one CVPR paper and one TMI paper about contrastive learning for semi-sueprvised segmentation. Please do keep an eye on this repository. If you have any interest about other methods, I am happy to try the code and add them to this repository. Thanks :-)
